src/netlink_namespace.c

```diff
@@ -0,0 +1,817 @@
+  if ((fields_num < 1) || (fields_num > 9)) {
+    sfree(new_val);
+    return (-1);
+  }
+
+  if ((strcasecmp(key, "Interface") == 0) ||
+      (strcasecmp(key, "VerboseInterface") == 0)) {
+    if (fields_num != 1) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`%s'. Got %i, expected 1.",
+            key, fields_num);
+      status = -1;
+    } else {
+      add_ignorelist(fields[0], "interface", NULL);
+      if (strcasecmp(key, "VerboseInterface") == 0)
+        add_ignorelist(fields[0], "if_detail", NULL);
+      status = 0;
+    }
+  } else if ((strcasecmp(key, "QDisc") == 0) ||
+             (strcasecmp(key, "Class") == 0) ||
+             (strcasecmp(key, "Filter") == 0)) {
+    if ((fields_num < 1) || (fields_num > 2)) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`%s'. Got %i, expected 1 or 2.",
+            key, fields_num);
+      return (-1);
+    } else {
+      add_ignorelist(fields[0], key, (fields_num == 2) ? fields[1] : NULL);
+      status = 0;
+    }
+  } else if (strcasecmp(key, "IgnoreSelected") == 0) {
+    if (fields_num != 1) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`IgnoreSelected'. Got %i, expected 1.",
+            fields_num);
+      status = -1;
+    } else {
+      if (IS_TRUE(fields[0]))
+        ir_ignorelist_invert = 0;
+      else
+        ir_ignorelist_invert = 1;
+      status = 0;
+    }
+  }
+  else if (strcasecmp(key, "NSName") == 0) {
+    INFO ("netlink_namespace plugin: recevied NSName value %s", fields[0]);
+    if ((fields_num < 1) || (fields_num > 2)) {
+      ERROR ("netlink_namespace plugin: Invalid number of fields for option "
+          "`%s'. Got %i, expected 1 or 2.", key, fields_num);
+      return (-1);
+    } else {
+      // ToDo: Check that /var/.... file exists (debuging messages please)
+      strncat(nspath, fields[0], sizeof(nspath) - strlen(nspath));
+      if (access(nspath, F_OK ) != 0 ) {
+        ERROR ("netlink_namespace plugin: ir_config: Unable to access file '%s'.", nspath);
+        status = -1;
+      } else {
+        INFO ("netlink_namespace plugin: ir_config: Setting config to %s", nspath);
+        ns_name = (char**) realloc(ns_name, sizeof(char *));
+        *ns_name = (char *) malloc(sizeof(char) * 1024);
+        strncpy(*ns_name, nspath, strlen(nspath));
+
+        INFO ("netlink_namespace pliugin: ir_config: ns_name is now %s", *ns_name);
+        
+        status = 0;
+      }
+    }
+  }
+
+  sfree(new_val);
+
+  return (status);
+} /* int ir_config */
+
+static int ir_init (void)
+{
+  DEBUG("Initializing netlink_namespace plugin\n");
+  if (ns_name != NULL) {
+    DEBUG("netlink_namespace plugin: ir_init: Switching to namespace on file '%s'.", *ns_name);
+
+    int fd = open(*ns_name, O_RDONLY);
+    if (fd  < 0) {
+        ERROR("Unable to open() %s", *ns_name);
+    }
+
+    if (setns(fd, CLONE_NEWNET) != 0) {
```

Comment:
It might actually be worth checking out if you can do something like this:

setns() -> change to namespace we want to monitor
get all the netlink sockets you need
setns() -> back to the network namespace we started in

I don't know if setns() changes what network namespace a socket()/file-descriptor is anchred in or not.